homeassistant/components/sensor/child_sensor.py

```diff
@@ -0,0 +1,144 @@
+"""
```

Comment:
So the problem that @jamespcole is trying to solve here is a new problem for Home Assistant: devices that have multiple functionalities that we _might_ want to expose.

I see child_sensor (and siblings) as derived entities, their behavior summarized as:
```javascript
f(state_parent_entity) = state child entity
```

I can see value for the child_**sensor** platform, albeit with a more descriptive name and very limited functionality. Only allow people to pass in `name`, `parent_entity_id`, `variable`, `unit_of_measurement`, `state_attributes`. I am afraid that adding more will only guide future developers to write more complex code instead of aiming for simplicity.

I think that the child_switch/child_light platforms are too confusing as they are sending data back to the parent. Making it very difficult to follow the data flow thus making it difficult to reason about what the system is doing when. It seems to me that they have been introduced to limit the calls to the Vera API. I think that doing this via states is the wrong approach. Instead, the Vera component should ensure it only makes 1 call per update and have the Vera entities read the cached result. [The Hue light component uses this approach.](https://github.com/balloob/home-assistant/blob/master/homeassistant/components/light/hue.py#L79-L106)

I think a more elegant solution to the fact that some entities can expose multiple functions is to follow an approach similar to the Z-Wave component. The Z-Wave component will take all sensor attributes from all devices and turn them into entities. It doesn't care which device exposes them. I think that Vera can benefit from the same approach:

 - Vera component will setup connection with the hub and load light/switch/sensor component _as needed_
 - sensor/vera will expose all sensor attributes on all Vera devices
 - switch/vera will expose each switch on each Vera device that should not be treated as a light
 - light/vera will expose all lights + switches that should be treated as lights